packages/freezed/test/common_types_test.dart

```diff
@@ -0,0 +1,231 @@
+// ignore_for_file: prefer_const_constructors, omit_local_variable_types, deprecated_member_use_from_same_package, cascade_invocations
+import 'package:analyzer/dart/analysis/results.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+import 'package:build_test/build_test.dart';
+import 'package:test/test.dart';
+
+import 'common.dart';
+import 'integration/common_types.dart';
+
+void main() {
+  test('has no issue', () async {
+    final main = await resolveSources(
+      {
+        'freezed|test/integration/common_types.dart': useAssetReader,
+      },
+      (r) => r.libraries.firstWhere(
+          (element) => element.source.toString().contains('common_types')),
+    );
+
+    final errorResult = await main.session
+            .getErrors('/freezed/test/integration/common_types.freezed.dart')
+        as ErrorsResult;
+
+    expect(errorResult.errors, isEmpty);
+  });
+
+  group('CommonSuperSubtype', () {
+    test('has the common properties available', () {
+      var value = CommonSuperSubtype('a', 1337, 42.24);
+      expect(value.a, 'a');
+      expect(value.b, 1337);
+      expect(value.c, 42.24);
+
+      value = CommonSuperSubtype.named('b', 12.34, 56.78);
+      expect(value.a, 'b');
+      expect(value.b, 12.34);
+      expect(value.c, 56.78);
+    });
+
+    test('copy has all with common subtypes available', () {
+      var value = CommonSuperSubtype('a', 1337, 42);
+      expect(value.a, 'a');
+
+      value = value.copyWith(a: 'b', c: 24.42);
+      expect(value.a, 'b');
+      expect(value.c, 24.42);
+    });
+
+    test(
+      'copy has shared params with common subtypes available',
+      () async {
+        await expectLater(compile(r'''
```

Comment:
For new tests checking compilation errors, it'd be great to use my expect_error package instead

This  `compile`  is technical dept and should be removed at some point.